test/SILOptimizer/discard_checking.swift

```diff
@@ -16,15 +16,15 @@ func globalThrowingFn() throws {}
 struct Basics: ~Copyable {
   consuming func test1(_ b: Bool) {
     guard b else {
-      fatalError("bah!") // expected-error {{must consume 'self' before exiting method that discards self}}
+      return // expected-error {{must consume 'self' before exiting method that discards self}}
```

Comment:
Curious why you needed to change these tests? I assume this fix revealed a bug in pruned liveness that the discard checking relied on w.r.t. unreachable blocks?

Reply:
These tests appear to previously be testing that an unreachable call like `fatalError` is treated as a function exit, which is wrong—`fatalError` never returns, and so the calling function is never really exited either. I needed to fix this because every force-unwrap introduces a branch with a trap and `unreachable` on the nil path, and if move checking tries to treat this as an exit, it will try to insert cleanups on the unreachable branch, which may not be valid because other cleanups on dependent accesses might still be active and we don't generally try to clean up on crash paths.